firmware/robot2015/src-ctrl/modules/control/RobotModel.hpp

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#define EIGEN_HAS_CXX11_MATH 0
+#include <Eigen/Dense>
+#include <array>
+#include "Geometry2d/Util.hpp"
+#include "const-math.hpp"
+
+/// Model parameters for a robot.  Used by the controls system.
+class RobotModel {
+public:
+    /// Radius of omni-wheel (in meters)
+    float WheelRadius;
+
+    /// Distance from center of robot to center of wheel
+    float WheelDist;
+
+    /// Wheel angles (in radians) measured between +x axis and wheel axle
+    std::array<float, 4> WheelAngles;
+
+    /// wheelSpeeds = BotToWheel * V_bot
+    Eigen::Matrix<float, 4, 3> BotToWheel;
+
+    /// This should be called when any of the other parameters are changed
+    void recalculateBotToWheel() {
+        // See this paper for more info on how this matrix is derived:
+        // http://people.idsia.ch/~foerster/2006/1/omnidrive_kiart_preprint.pdf
+
+        // clang-format off
+        BotToWheel <<
+            sinf(WheelAngles[0]), cosf(WheelAngles[0]), -WheelDist,
+            sinf(WheelAngles[1]), cosf(WheelAngles[1]), -WheelDist,
+            sinf(WheelAngles[2]), cosf(WheelAngles[2]), -WheelDist,
+            sinf(WheelAngles[3]), cosf(WheelAngles[3]), -WheelDist;
+        BotToWheel /= WheelRadius;
+        // clang-format on
+    }
+
+    /// (wheel rad/s desired) * DutyCycleMultiplier = duty cycle
+    /// Note that this is an approximation, as the relationship isn't exactly
+    /// linear
+    float DutyCycleMultiplier;
+};
+
+/// Model parameters for 2015 robot
+static const RobotModel RobotModel2015 = []() {
```

Comment:
Yeah you're right, `static` isn't what we want here.  http://stackoverflow.com/questions/92546/variable-declarations-in-header-files-static-or-not

I did it this way as opposed to using a constructor for all of the parameters for readability and less code.  This way I don't have to write a constructor that does `x = x` for the 5 params.  It also is easier to read (I think) the initialization parameters this way as opposed to a constructor that takes in 4 floats and an array - it's hard to tell what's what.